src/components/dropdownSearch/DropdownSearchReducers.tsx

```diff
@@ -50,27 +50,84 @@ export const getNextIndexPosition = (array: any[], item: any, key: number): numb
+  if (selectedOption) {
+    if (selectedOption.custom) {
+      nextOptions.splice(nextOptions.indexOf(selectedOption), 1);
+    } else {
+      selectedOption.selected = false;
+      selectedOption.hidden = false;
+    }
+  }
+  return nextOptions;
+};
+
+export const removeLastSelectedOption = (options: IDropdownOption[]): IDropdownOption[] => {
+  const nextOptions: IDropdownOption[] = deepClone(options);
+  const lastSelectedOption: IDropdownOption = _.find(nextOptions.reverse(), { selected: true });
+  if (lastSelectedOption) {
+    return removeSelectedOption(options, lastSelectedOption.displayValue);
+  } else {
+    return nextOptions.reverse();
+  }
+};
 
-export const addUniqueSelectedOption = (state: IDropdownSearchState, displayValue: string): FixedQueue<IDropdownOption> =>
-  removeSelectedOption(state, displayValue).immutablePush({ value: UUID.generate(), displayValue });
+export const removeAllSelectedOption = (options: IDropdownOption[]): IDropdownOption[] => {
+  return _.map(options, (option: IDropdownOption) => {
+    const nextOption = deepClone(option);
+    nextOption.selected = false;
+    nextOption.hidden = false;
+    return nextOption;
+  });
+};
 
-export const getDisplayedOptions = (state: IDropdownSearchState) => {
-  return _.filter(state.options,
-    (option: IDropdownOption) => _.findWhere(state.selectedOptions.getQueue(),
-      { displayValue: option.displayValue }) === undefined);
+export const addUniqueSelectedOption = (options: IDropdownOption[], displayValue: string): IDropdownOption[] => {
+  const nextOptions: IDropdownOption[] = deepClone(options);
+  nextOptions.push({
+    displayValue,
+    value: UUID.generate(),
+    selected: true,
+    custom: true,
+  });
+  return nextOptions;
+};
+
+export const getDisplayedOptions = (options: IDropdownOption[]): IDropdownOption[] => {
+  return _.reject(options, (option) => {
+    return option.custom || option.hidden;
+  });
 };
 
 export const getFilteredOptions = (state: IDropdownSearchState, filterText?: string) => {
   const currentFilterText: string = filterText || state.filterText;
-  return _.filter(getDisplayedOptions(state),
+  return _.filter(getDisplayedOptions(state.options),
     (option: IDropdownOption) => {
       const value = option.displayValue || option.value;
       return _.isEmpty(currentFilterText) || s.contains(value.toLowerCase(), (currentFilterText).toLowerCase());
     });
 };
 
+export const selectSingleOption = (options: IDropdownOption[], selectedOption: IDropdownOption): IDropdownOption[] => {
+  return _.map(options, (option: IDropdownOption) => {
+    const nextOption = deepClone(option);
+    nextOption.selected = option.value === selectedOption.value;
+    return nextOption;
+  });
+};
+
+export const multiSelectOption = (options: IDropdownOption[], selectedOption: IDropdownOption): IDropdownOption[] => {
+  return _.map(options, (option: IDropdownOption) => {
```

Comment:
The option needs to be hidden when multi-selected in this case so your proposition would not work here.